include/benchmark/benchmark.h

```diff
@@ -465,7 +465,13 @@ inline BENCHMARK_ALWAYS_INLINE void DoNotOptimize(Tp const& value) {
 }
 
 template <class Tp>
-inline BENCHMARK_ALWAYS_INLINE void DoNotOptimize(Tp& value) {
+inline BENCHMARK_ALWAYS_INLINE void DoNotOptimize(
+#ifdef BENCHMARK_HAS_CXX11
```

Comment:
we have a BENCHMARK_HAS_CXX11 block below.  can we not simplify this somehow so this is folded into the version at line 510?

Reply:
I think about this, but in case of reordering blocks, the behaviour will be changed.

on other hand, if the BENCHMARK_HAS_CXX11 related code is moved to separate block - it causes code duplication (